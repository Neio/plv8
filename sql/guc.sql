-- enable strict mode with GUC, which allows usage of ES6 let and const keywords
SET plv8.v8_flags = '--use_strict';
CREATE OR REPLACE FUNCTION let_test()
   RETURNS json AS $$
      let result = ['Hello, World!'];
      return result;
   $$ LANGUAGE plv8 STABLE STRICT;
SELECT let_test();
DROP FUNCTION let_test();

-- set and show variables
SET plv8.start_proc = 'foo';
SHOW plv8.start_proc;

SET plv8.icu_data = 'bar';
SHOW plv8.icu_data;

SET plv8.v8_flags = 'baz';
SHOW plv8.v8_flags;
