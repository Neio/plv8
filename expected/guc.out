-- enable strict mode with GUC, which allows usage of ES6 let and const keywords
SET plv8.v8_flags = '--use_strict';
CREATE OR REPLACE FUNCTION let_test()
   RETURNS json AS $$
      let result = ['Hello, World!'];
      return result;
   $$ LANGUAGE plv8 STABLE STRICT;
SELECT let_test();
     let_test      
-------------------
 ["Hello, World!"]
(1 row)

DROP FUNCTION let_test();
-- set and show variables
SET plv8.start_proc = 'foo';
SHOW plv8.start_proc;
 plv8.start_proc 
-----------------
 foo
(1 row)

SET plv8.icu_data = 'bar';
SHOW plv8.icu_data;
 plv8.icu_data 
---------------
 bar
(1 row)

SET plv8.v8_flags = 'baz';
SHOW plv8.v8_flags;
 plv8.v8_flags 
---------------
 baz
(1 row)

